### PR TITLE
Introduce Pinia store to manage app state

### PR DESCRIPTION
**Pull request by pranavxc**
## Change Summary

- [x] Use pinia for app state management
- [x] Move `useProject` composable to Pinia store

    #### Syntax to use `useProject` store
    
    ```ts
    // get project store reference
    const projectStore = useProject()
    
    // get actions from store as usual
    const { loadTables } = projectStore
    
    // get  any state values or getters to reactive variable
    const { project } = storeToRefs(projectStore)
    // alternately it can be directly used from store like `projectStore.project`
    ```
- [x] Move `useTabs` store to Pinia store

## Change type

- [x] feat: (new feature for the user, not a new feature for build script)
- [ ] fix: (bug fix for the user, not a fix to a build script)
- [ ] docs: (changes to the documentation)
- [ ] style: (formatting, missing semi colons, etc; no production code change)
- [x] refactor: (refactoring production code, eg. renaming a variable)
- [ ] test: (adding missing tests, refactoring tests; no production code change)
- [ ] chore: (updating grunt tasks etc; no production code change)

## Test/ Verification

Provide summary of changes.

## Additional information / screenshots (optional)

Anything for maintainers to be made aware of

### CODE REVIEW

File: packages/nc-gui/components/dashboard/settings/Misc.vue [hunk 1]
```diff
@@ -4,7 +4,9 @@ import { useGlobal, useProject, watch } from '#imports'
 
 const { includeM2M, showNull } = useGlobal()
 
-const { project, updateProject, projectMeta, loadTables, hasEmptyOrNullFilters } = useProject()
+const projectStore = useProject()
+const { updateProject, loadTables, hasEmptyOrNullFilters } = projectStore
+const { project, projectMeta } = projectStore
```
Comment (wingkwong) [inline packages/nc-gui/components/dashboard/settings/Misc.vue:9]: missing `storeToRefs`

File: packages/nc-gui/components/dashboard/settings/DataSources.vue [hunk 2]
```diff
@@ -22,7 +22,9 @@ const vState = useVModel(props, 'state', emits)
 const vReload = useVModel(props, 'reload', emits)
 
 const { $api, $e } = useNuxtApp()
-const { project, loadProject } = useProject()
+const projectStore = useProject()
+const { loadProject } = projectStore
+const { project } = projectStore
```
Comment (wingkwong) [inline packages/nc-gui/components/dashboard/settings/DataSources.vue:27]: missing `storeToRefs`

File: packages/nc-gui/components/dashboard/settings/data-sources/EditBase.vue [hunk 3]
```diff
@@ -31,7 +31,9 @@ const props = defineProps<{
 
 const emit = defineEmits(['baseUpdated'])
 
-const { project, loadProject } = useProject()
+const projectStore = useProject()
+const { loadProject } = projectStore
+const { project } = projectStore
```
Comment (wingkwong) [inline packages/nc-gui/components/dashboard/settings/data-sources/EditBase.vue:36]: missing `storeToRefs`

File: packages/nc-gui/components/dlg/AirtableImport.vue [hunk 4]
```diff
@@ -31,7 +31,9 @@ const baseURL = appInfo.ncSiteUrl
 
 const { $state } = useNuxtApp()
 
-const { project, loadTables } = useProject()
+const projectStore = useProject()
+
+const { project } = projectStore
```
Comment (wingkwong) [inline packages/nc-gui/components/dlg/AirtableImport.vue:36]: missing `storeToRefs`

File: packages/nc-gui/components/dlg/TableRename.vue [hunk 5]
```diff
@@ -38,7 +38,9 @@ const dialogShow = useVModel(props, 'modelValue', emit)
 
 const { updateTab } = useTabs()
 
-const { loadTables, tables, project, isMysql, isMssql, isPg } = useProject()
+const projectStore = useProject()
+const { loadTables, isMysql, isMssql, isPg } = projectStore
+const { project } = projectStore
```
Comment (wingkwong) [inline packages/nc-gui/components/dlg/TableRename.vue:43]: missing `storeToRefs`

File: packages/nc-gui/components/smartsheet/sidebar/index.vue [hunk 6]
```diff
@@ -21,11 +22,11 @@ const meta = inject(MetaInj, ref())
 
 const activeView = inject(ActiveViewInj, ref())
 
-const { activeTab } = useTabs()
+const { activeTab } = storeToRefs(useTabs())
 
 const { views, loadViews, isLoading } = useViews(meta)
```
Comment (wingkwong) [inline packages/nc-gui/components/smartsheet/sidebar/index.vue:27]: should apply `storeToRefs` for views & isLoading

File: packages/nc-gui/composables/useMultiSelect/index.ts [hunk 7]
```diff
@@ -52,7 +53,7 @@ export function useMultiSelect(
 
   const { appInfo } = useGlobal()
 
-  const { isMysql } = useProject()
+  const { isMysql } = storeToRefs(useProject)
```
Comment (wingkwong) [inline packages/nc-gui/composables/useMultiSelect/index.ts:56]: `isMysql` is a function. no need to use `storeToRefs`

File: packages/nc-gui/store/tab.ts [hunk 8]
```diff
@@ -10,14 +11,14 @@ function getPredicate(key: Partial<TabItem>) {
...
-export const useTabs = createSharedComposable(() => {
+export const useTabs = defineStore('tabStore', () => {
   const tabs = ref<TabItem[]>([])
 
   const router = useRouter()
 
   const route = $(router.currentRoute)
 
-  const { bases, tables } = useProject()
+  const projectStore = useProject()
```
Comment (wingkwong) [inline packages/nc-gui/store/tab.ts:21]: need `storeToRefs` for `bases` and `tables`

File: packages/nc-gui/components/cell/SingleSelect.vue [hunk 9]
```diff
@@ -60,7 +61,7 @@ const { getMeta } = useMetas()
 
 const { hasRole } = useRoles()
 
-const { isPg, isMysql } = useProject()
+const { isPg, isMysql } = storeToRefs(useProject())
```
Comment (wingkwong) [inline packages/nc-gui/components/cell/SingleSelect.vue:64]: they are functions. `storeToRefs` is not required.

**Review by wingkwong — CHANGES_REQUESTED** [review]

File: packages/nc-gui/store/tab.ts (hunk 8 above)
Comment (pranavxc) [reply packages/nc-gui/store/tab.ts:21]: we are directly using it from `projectStore`

File: packages/nc-gui/components/smartsheet/sidebar/index.vue (hunk 6 above)
Comment (pranavxc) [reply packages/nc-gui/components/smartsheet/sidebar/index.vue:27]: Both `loadViews, isLoading` are loading from `useViews` which is not a Pinia store

**Review by wingkwong — COMMENTED** [review]
would the existing eventBus logic also migrate to pinia in this pr? e.g. Menu.vue.

**Comment by pranavxc** [comment]
> would the existing eventBus logic also migrate to pinia in this pr? e.g. Menu.vue.

It's not migrated yet. At the moment just migrated the tab and project state to Pinia. Those refactoring we can do later.

**Review by wingkwong — APPROVED** [review]

**Pull request merged**